File: Algorithms/CountSort/CountSort.py
def count_sort(imput_array, values_range, size_array):
    count_array = []
    output_array = []

    for i in range(values_range):
        count_array.append(0)

    for i in range(size_array):
        output_array.append(0)
        count_array[imput_array[i] - 1] = count_array[imput_array[i] - 1] + 1

    for i in range(1, values_range, 1):
        count_array[i] = count_array[i] + count_array[i - 1]

    for i in range(size_array - 1, -1, -1):
        output_array[count_array[imput_array[i] - 1] - 1] = imput_array[i]
        count_array[imput_array[i] - 1] = count_array[imput_array[i] - 1] - 1

    return output_array


def main():
    size_array = int(input())
    array = [int(i) for i in input().split()]

    output = count_sort(array, 10, size_array)

    for i in range(size_array):
        if i < size_array-1:
            print(output[i], end=' ')
        else:
            print(output[i])


if __name__ == '__main__':
    main()

File: Algorithms/CountSort/CountSort.cpp
#include <algorithm>
#include <iostream>
#include <vector>

  
void countSort(std::vector<int>& inputArray) {
    int maxElement = *max_element(inputArray.begin(), inputArray.end());
    int minElement = *min_element(inputArray.begin(), inputArray.end());
    int range = maxElement - minElement + 1;
  
    std::vector<int> countArray(range), outputArray(inputArray.size());

    for (int i = 0; i < inputArray.size(); i++) {
        countArray[inputArray[i] - minElement]++;
    }
  
    for (int i = 1; i < countArray.size(); i++) {
        countArray[i] += countArray[i - 1];
    }
  
    for (int i = inputArray.size() - 1; i >= 0; i--) {
        outputArray[countArray[inputArray[i] - minElement] - 1] = inputArray[i];
        countArray[inputArray[i] - minElement]--;
    }
  
    for (int i = 0; i < inputArray.size(); i++) {
        inputArray[i] = outputArray[i];
    }
}


void printArray(std::vector<int>& inputArray) {
    for (int i = 0; i < inputArray.size(); i++)
        std::cout << inputArray[i] << " ";
    std::cout << std::endl;
}


int main()
{
    size_t sizeArray;
    std::cin >> sizeArray;

    std::vector<int> array(sizeArray);
    for (int &element : array) {
        std::cin >> element;
    }

    countSort(array);
    printArray(array);
    return 0;
}